File: Typescript/Practice/26072022/medium/medium_functionFactory.ts
#!/usr/bin/env npx ts-node

function makePlusFunction (num: number) {
  let func = function (num2:number): number {
    return num + num2;
  };
  return func;
};

const plusFive = makePlusFunction(5);

console.log(plusFive(2));
console.log(plusFive(-8));

const plusTen = makePlusFunction(10);

console.log(plusTen(0));
console.log(plusTen(188));
console.log(plusFive(plusTen(0)));

File: Typescript/Practice/medium/medium_functionFactory.py
#!/usr/bin/env python

def make_plus_function(num):
  def func(num2):
    return num + num2
  return func

plus_five = make_plus_function(5)

print(plus_five(2))
print(plus_five(-8))

plus_ten = make_plus_function(10)

print(plus_ten(0))
print(plus_ten(188))
print(plus_five(plus_ten(0)))
